### DISCUSSION TITLE
PnPContext RestClient

### DISCUSSION THREAD

**Discussion by anthonywhite** [pnp/pnpframework#294] [Q&A] [answered]
Is the [PnPContext.RestClient](https://pnp.github.io/pnpcore/api/PnP.Core.Services.PnPContext.html#PnP_Core_Services_PnPContext_RestClient) intended to operate with the same OAuth context as its `PnPContext`?

When I try to use it I get a 403 even though the `PnPContext` works fine.

**Comment by jansenbe** [comment, marked as answer]
@anthonywhite : it's just the http client, the oauth bearer token is added internally in PnP Core SDK. What would be the API call you wanted to make? Ideally we extend PnP Core SDK so you make the call in a "typed" manner

**Reply by anthonywhite** [reply]
Thanks Bert, API call I need - create a non-group-connected Site collection, as discussed in [my other Q&A](https://github.com/pnp/pnpframework/discussions/289) ;-). When you say this:

> the oauth bearer token is added internally in PnP Core SDK

...do you mean that should have happened when I set up the PnPContext, or is there something else I need to do?

**Reply by jansenbe** [reply]
You could do something like below. I'm looking into making this approach easier going forward, stay tuned for an update

```csharp
ctx.RestClient.Client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("bearer", await ctx.AuthenticationProvider.GetAccessTokenAsync("https://contoso.sharepoint.com").ConfigureAwait(false));
```

**Reply by jansenbe** [reply]
See https://pnp.github.io/pnpcore/using-the-sdk/basics-pnpframework.html for documentation on PnP Framework / PnP Core SDK Interop

**Reply by anthonywhite** [reply]
Bert, picked up last night's build and all working great! How nice to come back from a UK holiday weekend and find this all done. The weights on the PnP heavy lifting bar getting steadily bigger...

Many Thanks again for looking at this so quickly. 👍🏻
